### PR TITLE
Fix returning success from DeleteSnapshot for stale requests

### PR DESCRIPTION
**Pull request by ShyamsundarR**
Also reduced code duplication in fetching pool list from Ceph.

DeleteSnapshot like DeleteVolume, should return a success when it
detects that the snapshot keys are missing from the RADOS OMaps that
store the snapshot UUID to request name mapping.

This was missing in the code, and is now added.

Signed-off-by: ShyamsundarR <srangana@redhat.com>

### CODE REVIEW

File: pkg/util/cephcmds.go [hunk 1]
```diff
@@ -47,15 +47,14 @@ func ExecCommand(program string, args ...string) (stdout, stderr []byte, err err
 	return stdoutBuf.Bytes(), nil, nil
 }
 
-// cephStoragePoolSummary strongly typed JSON spec for osd ls pools output
-type cephStoragePoolSummary struct {
+// CephStoragePoolSummary strongly typed JSON spec for osd ls pools output
+type CephStoragePoolSummary struct {
```
Comment (humblec) [inline pkg/util/cephcmds.go]: @ShyamsundarR why we need above change? afaict, its used internally here .

Comment (ShyamsundarR) [reply pkg/util/cephcmds.go]: static linters complained that this was bad style, hence fixed it up.

Comment (humblec) [reply pkg/util/cephcmds.go]: We dont need to export a type unwantedly..

Comment (Madhu-1) [reply pkg/util/cephcmds.go]: @humblec if you don't fix it CI will  fail

Comment (Madhu-1) [reply pkg/util/cephcmds.go]: this function may get  called by other functions  in  different package  hence it's required

File: pkg/util/cephcmds.go [hunk 2]
```diff
@@ -47,15 +47,14 @@ func ExecCommand(program string, args ...string) (stdout, stderr []byte, err err
...
 	Name   string `json:"poolname"`
 	Number int64  `json:"poolnum"`
 }
 
-// GetPoolID searches a list of pools in a cluster and returns the ID of the pool that matches
-// the passed in poolName parameter
-func GetPoolID(monitors string, cr *Credentials, poolName string) (int64, error) {
+// GetPools fetches a list of pools from a cluster
+func GetPools(monitors string, cr *Credentials) ([]CephStoragePoolSummary, error) {
```
Comment (Madhu-1) [inline pkg/util/cephcmds.go]: why this function is exported? is this expected to be used in other packages?

Comment (ShyamsundarR) [reply pkg/util/cephcmds.go]: It is currently not called form other packages, one way to possibly fix the other comment on the linter problem is to not expose this. Is that what you want?

Comment (Madhu-1) [reply pkg/util/cephcmds.go]: yes, unexporting this function should fix it.

File: pkg/util/cephcmds.go (hunk 1 above)
Comment (ShyamsundarR) [reply pkg/util/cephcmds.go]: Fixed.

File: pkg/util/cephcmds.go (hunk 2 above)
Comment (ShyamsundarR) [reply pkg/util/cephcmds.go]: Fixed.

File: pkg/util/cephcmds.go (hunk 1 above)
Comment (humblec) [reply pkg/util/cephcmds.go]: @Madhu-1 if it was used in other packages, this wanted to be "exported" before this PR itself, Isnt it ? :).

Comment (humblec) [reply pkg/util/cephcmds.go]: Thanks for fixing it @ShyamsundarR

File: pkg/util/cephcmds.go [hunk 3]
```diff
@@ -69,14 +68,25 @@ func GetPoolID(monitors string, cr *Credentials, poolName string) (int64, error)
...
-		return 0, err
+		return nil, err
 	}
 
 	var pools []cephStoragePoolSummary
 	err = json.Unmarshal(stdout, &pools)
 	if err != nil {
 		klog.Errorf("failed to parse JSON output of pool list from cluster (%s)", err)
-		return 0, fmt.Errorf("unmarshal failed: %+v.  raw buffer response: %s", err, string(stdout))
+		return nil, fmt.Errorf("unmarshal failed: %+v.  raw buffer response: %s", err, string(stdout))
```
Comment (humblec) [inline pkg/util/cephcmds.go]: @ShyamsundarR may be "unmarshal of pool list failed..." looks more informative.

Comment (ShyamsundarR) [reply pkg/util/cephcmds.go]: Done.

**Pull request merged**